### DISCUSSION TITLE
What actually happening when we run `forge test`?

### DISCUSSION THREAD

**Discussion by TheMj0ln1r** [Cyfrin/foundry-full-course-cu#362] [Q&A] [answered]
We are testing the contract on the MAINNET by `fork test --fork-url $MAINNET_URL`. Why it doesn't cost us the gas? Aren't we are actually on the MAINNET?

**Comment by Naman1729** [comment, marked as answer]
Hey @TheMj0ln1r , During a fork test, the code execution and transactions occur locally on a **simulated blockchain instance**, replicating the state of the Mainnet. However, these transactions are not broadcasted or executed on the _actual Mainnet network_. As a result, no gas fees are incurred because the interactions are confined to the local testing environment.

for more information go to this link:
https://dev.to/vishakbhaskar/fork-mainnet-using-hardhat-to-test-and-build-on-defi-protocols-and-more-5a33

**Reply by TheMj0ln1r** [reply]
Thank you for the confirmation mate!
